### PR TITLE
use C Macro to register singletons

### PR DESCRIPTION
**Pull request by 3nids**
This will avoid writing repetitive code in the source file

### CODE REVIEW

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-7c926ec229e50b4402af6263cf5018a967e0ac4d-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-7c926ec229e50b4402af6263cf5018a967e0ac4d-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-7c926ec229e50b4402af6263cf5018a967e0ac4d-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-7c926ec229e50b4402af6263cf5018a967e0ac4d-x86_64.apk)

File: src/core/qgismobileapp.h [hunk 1]
```diff
@@ -52,6 +52,11 @@ class LegendImageProvider;
 class QgsProject;
 
 
+#define SINGLETON_PROVIDER(_class) static QObject * _class##SingletonProvider(QQmlEngine *engine, QJSEngine *scriptEngine) {Q_UNUSED(engine); Q_UNUSED(scriptEngine); return new _class();}
+#define REGISTER_SINGLETON(uri, _class, name) qmlRegisterSingletonType<_class>( uri, 1, 0, name, _class##SingletonProvider )
```
Comment (m-kuhn) [inline src/core/qgismobileapp.h]:
```suggestion
#define REGISTER_SINGLETON(uri, _class, name) qmlRegisterSingletonType<_class>( uri, 1, 0, name, [] ( QQmlEngine *engine, QJSEngine *scriptEngine ) -> QObject * { Q_UNUSED(engine); Q_UNUSED(scriptEngine); return new _class(); } )
```

Comment (m-kuhn) [reply src/core/qgismobileapp.h]: That will require no modification of the header at all

**Comment by qfield-fairy** [comment]
Uploaded test apks for
 - [**armv7**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-2ce55faad6b613111483f7e8563747c42b9fafdf-armv7.apk)
 - [**x86**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-2ce55faad6b613111483f7e8563747c42b9fafdf-x86.apk)
 - [**arm64_v8a**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-2ce55faad6b613111483f7e8563747c42b9fafdf-arm64_v8a.apk)
 - [**x86_64**](https://download.opengis.ch/qfield/ci-builds/qfield-dev-813-2ce55faad6b613111483f7e8563747c42b9fafdf-x86_64.apk)

**Pull request merged**